src/main/java/uk/gov/pay/ledger/transaction/dao/TransactionDao.java

```diff
@@ -20,7 +21,9 @@
 
     private static final String FIND_TRANSACTION_BY_EXTERNAL_ID_AND_GATEWAY_ACCOUNT_ID = "SELECT * FROM transaction " +
             "WHERE external_id = :externalId " +
-            "and gateway_account_id = :gatewayAccountId";
+            "AND gateway_account_id = :gatewayAccountId " +
+            "AND (:transactionType::transaction_type is NULL OR type = :transactionType::transaction_type) " +
```

Comment:
Makes me worried why this change didn't break existing pacts. For `a transaction with created state exist` we don't currently pass transactionType parameter and as the query should now be `transaction_type is NULL` (which is not currently) I am expecting pact test to break unless I am missing something ?

Reply:
Looks like we don't have a pact test for `get-one-transaction`

Reply:
`:transactionType::transaction_type is NULL` checks whether the passed parameter is null, not the value of the column in db.
For Payment we don't currently pass value for transaction type when we retrieve it in public api from ledger. (It will be done separately)

Reply:
I prefer constructing query if value is available. Not a great fan of using `NULL is NULL` condition but fine with it too.